eng/test-job.yml

```diff
@@ -323,6 +323,26 @@ jobs:
           - forcerelocs
           - gcstress15
 
+    # Zip tests
+    - ${{ if ne(parameters.osGroup, 'Windows_NT') }}:
+      - script: pushd $(Build.SourcesDirectory)/bin/tests/$(osGroup).$(archType).$(buildConfigUpper) && tar -czvf $(Build.SourcesDirectory)/bin/tests/$(osGroup).$(archType).$(buildConfigUpper).tar.gz ./* && popd
+        displayName: Tar Tests
+    - ${{ if eq(parameters.osGroup, 'Windows_NT') }}:
+      - script: powershell -NoProfile -Command "Add-Type -Assembly 'System.IO.Compression.FileSystem'; [System.IO.Compression.ZipFile]::CreateFromDirectory('$(Build.SourcesDirectory)\\bin\\tests\\$(osGroup).$(archType).$(buildConfigUpper)', '$(Build.SourcesDirectory)\\bin\\tests\\$(osGroup).$(archType).$(buildConfigUpper).zip')"
+        displayName: Zip Tests
+
+    - ${{ if ne(parameters.osGroup, 'Windows_NT') }}:
+      - task: PublishBuildArtifacts@1
+        displayName: Publish Test Build
+        inputs:
+          pathtoPublish: $(Build.SourcesDirectory)/bin/tests/$(osGroup).$(archType).$(buildConfigUpper).tar.gz
+          artifactName: ${{ format('{0}_{1}_{2}_tests', parameters.osIdentifier, parameters.archType, parameters.buildConfig) }}
```

Comment:
@AaronRobinsonMSFT is working on renaming our artifacts. Maybe he can comment abou the new naming scheme he has in mind.

Reply:
Thanks @echesakov . I would prefer to put the `Tests_` prefix first instead of last. I am also updating the logging to have the `TestLogs_` prefix. You can also merge this in and I can update it when I resolve conflicts.

Reply:
Will change.